app/views/courses/show.haml

```diff
@@ -61,3 +61,22 @@
             %td= link_to assignment.name, [@course, assignment]
             %td= assignment.due_date
 
+= panel title: 'Assigment results', context: :primary do
+  .panel-body
+    %table.table.table-striped
+      %thead
+        %tr
+          %th= 'Student'
+          - @assignments.decorate.each do |assignment|
+            %th= link_to assignment.name, [@course, assignment]
+      %tbody
+        - @course.student_memberships.each do |student|
+          %tr
+            %td
+              = student.name
+            - @assignments.decorate.each do |assignment|
+              %td
+                - if assignment.submissions.pluck('uploaded_by_id').include? student.user_id # validate that student made submission
```

Comment:
To make it more object oriented (and performant), how about creating a method in assignment? `submission_for(student)`. This makes the code a lot cleaner, and you don't have to do as many DB queries